src/MagnumPlugins/CgltfImporter/CgltfImporter.cpp

```diff
@@ -0,0 +1,2562 @@
+        meshPrimitive = MeshPrimitive::TriangleStrip;
+    } else {
+        /* Cgltf parses an int and directly casts it to cgltf_primitive_type
+           without checking for valid values */
+        Error{} << "Trade::CgltfImporter::mesh(): unrecognized primitive" << primitive.type;
+        return Containers::NullOpt;
+    }
+
+    /* Sort attributes by name so that we add attribute sets in the correct
+       order and can warn if indices are not contiguous. Stable sort is needed
+       to preserve declaration order for duplicate attributes, checked below. */
+    Containers::Array<UnsignedInt> attributeOrder{primitive.attributes_count};
+    for(UnsignedInt i = 0; i < attributeOrder.size(); ++i)
+        attributeOrder[i] = i;
+
+    std::stable_sort(attributeOrder.begin(), attributeOrder.end(), [&](UnsignedInt a, UnsignedInt b) {
+            return std::strcmp(primitive.attributes[a].name, primitive.attributes[b].name) < 0;
+        });
+
+    /* Find and remove duplicate attributes. This mimics tinygltf behaviour
+       which replaces the previous attribute of the same name. */
+    std::size_t attributeCount = attributeOrder.size();
+    for(UnsignedInt i = 0; i + 1 < attributeOrder.size(); ++i) {
+        const cgltf_attribute& current = primitive.attributes[attributeOrder[i]];
+        const cgltf_attribute& next = primitive.attributes[attributeOrder[i + 1]];
+        if(std::strcmp(current.name, next.name) == 0) {
+            --attributeCount;
+            /* Mark for skipping later */
+            attributeOrder[i] = ~0u;
+        }
+    }
+
+    /* Gather all (whitelisted) attributes and the total buffer range spanning
+       them */
+    cgltf_buffer* buffer = nullptr;
+    UnsignedInt vertexCount = 0;
+    std::size_t attributeId = 0;
+    cgltf_attribute lastAttribute{};
+    Math::Range1D<std::size_t> bufferRange;
+    Containers::Array<MeshAttributeData> attributeData{attributeCount};
+    for(UnsignedInt a: attributeOrder) {
+        /* Duplicate attribute, skip */
+        if(a == ~0u)
+            continue;
+
+        const cgltf_attribute& attribute = primitive.attributes[a];
+
+        const Containers::StringView nameString{attribute.name};
+        /* See the comment in doOpenData() for why we do this */
+        const Containers::StringView semantic = attribute.type != cgltf_attribute_type_invalid ?
+            nameString.partition('_')[0] : nameString;
+
+        if(attribute.type != cgltf_attribute_type_invalid) {
+            if(attribute.type != lastAttribute.type) {
+                if(attribute.index != 0)
+                    Warning{} << "Trade::CgltfImporter::mesh(): expected attribute" << nameString << "to start at index 0";
+            } else if(attribute.index != lastAttribute.index + 1)
+                Warning{} << "Trade::CgltfImporter::mesh(): expected attribute" << nameString << "to continue at index" << lastAttribute.index + 1;
+        }
```

Comment:
What about 

```suggestion
        if(attribute.type != cgltf_attribute_type_invalid && ((attribute.type != lastAttribute.type && attribute.index != 0) || attribute.index != lastAttribute.index + 1))
            Warning{} << "Trade::CgltfImporter::mesh(): found attribute" << nameString << "but there was no" << Utility::formatString("{}_{}", semantic, attribute.index - 1);
```

which would print e.g. `Trade::CgltfImporter::mesh(): found attribute COLORS_3 but there was no COLORS_2`? I can understand that warning a lot better.